common/oauthTokenManager.go

```diff
@@ -617,7 +568,7 @@ func (credInfo *OAuthTokenInfo) GetManagedIdentityCredential() (azcore.TokenCred
 	return tc, nil
 }
 
-func (credInfo *OAuthTokenInfo) GetClientCertificateCredential() (azcore.TokenCredential, error) {
+func (credInfo *OAuthTokenInfo) GetClientCertificateCredential(persist bool) (azcore.TokenCredential, error) {
```

Comment:
Instead of passing persist to each method, we should store it as part of OAuthTokenInfo

Reply:
Done in the latest commit.